go/vt/vtctl/reparentutil/planned_reparenter.go

```diff
@@ -200,6 +203,20 @@ func (pr *PlannedReparenter) preflightChecks(
 		return true, vterrors.Errorf(vtrpc.Code_FAILED_PRECONDITION, "primary-elect tablet %v is not in the shard", primaryElectAliasStr)
 	}
 
+	// We check for replication lag if the primary was provided to us and tolerable replication lag is defined.
+	if toCheckReplicationLag && opts.TolerableReplLag > 0 {
+		// Find the replication lag for the primary-elect tablet.
+		_, replLag, err := findPositionAndLagForTablet(ctx, newPrimaryTabletInfo.Tablet, pr.logger, pr.tmc, opts.WaitReplicasTimeout)
+		if err != nil {
+			return true, err
+		}
```

Comment:
We are duplicating code here that is also in `ChooseNewPrimary`. Is there a way to avoid that?
Maybe we could push the case where `NewPrimaryAlias` is already provided also into `ChooseNewPrimary`?

Reply:
Looks like @maxenglander had a similar reaction, with a more concrete suggestion.